addons/godot-next/objects/message_dispatcher.gd

```diff
@@ -0,0 +1,51 @@
+# MessageDispatcher
+# author: MunWolf (Rikhardur Bjarni Einarsson)
+# license: MIT
+# copyright: Copyright (c) 2019 Rikhardur Bjarni Einarsson
+# brief_description: A message handler for non predefined signals, can be extended by any class
+#                    Node, Reference, Resource etc, can also be directly attached to a Node.
+extends Object
```

Comment:
This object should probably extend Reference rather than Object. I don't really see any reason why the users should need to keep track of where this object should be deleted.

Reply:
I had it object because Node does not inherit from Reference.

Reply:
I could also just provide a Node wrapper for this that people could extend instead of ectending the MessageDispatcher directly, that would mean I could use Reference here.